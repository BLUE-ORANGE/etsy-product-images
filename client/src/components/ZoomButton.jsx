import React from 'react';
import ZoomedPhoto from './ZoomedPhoto';

class ZoomButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displayZoom: false,
      imageUrl: 'https://i.imgur.com/ZgZ8YKN.jpg',
    };
  }

  handleClick(e) {
    console.log(e.target);
    this.setState({
      displayZoom: true,
    });

  }

  render() {
    return (
      <div>
        <div onClick={ (e) => this.handleClick(e) }>
          <span className="ss-icon">ðŸ”Ž</span>zoom
        </div>
        <div>
          {
          this.state.displayZoom ? <ZoomedPhoto imageUrl={this.state.imageUrl} /> : ''
        }
        </div>
      </div>
    );
  }
}


export default ZoomButton;
