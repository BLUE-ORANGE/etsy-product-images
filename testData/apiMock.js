module.exports.testData = [
  {
    id: 108,
    imageUrl: 'https://i.imgur.com/zmTGRu1.jpg',
    focused: true,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  }, {
    id: 247,
    imageUrl: 'https://i.imgur.com/V1XNLjh.jpg',
    focused: false,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  }, {
    id: 731,
    imageUrl: 'https://i.imgur.com/jpc5JTA.jpg',
    focused: false,
    productId: 108,
    createdAt: '2018-02-26T00:17:52.802Z',
    updatedAt: '2018-02-26T00:17:52.802Z',
  },
];
