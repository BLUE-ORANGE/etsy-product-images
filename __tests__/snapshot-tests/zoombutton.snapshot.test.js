import React from 'react';
import renderer from 'react-test-renderer';
import ZoomButton from '../../client/src/components/ZoomButton';
import testData from '../../testData/apiMock';

it('renders the zoom button with reference to imageUrl', () => {
  const tree = renderer
    .create(<ZoomButton
      medium={testData.testData[0].imageUrl}
      large={testData.testData[0].imageUrl}
    />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

