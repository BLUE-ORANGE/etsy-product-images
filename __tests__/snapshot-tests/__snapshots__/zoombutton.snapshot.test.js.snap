// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the zoom button with reference to imageUrl 1`] = `
<a
  className="zoom"
  id="zoom"
  onClick={[Function]}
>
  <span
    className="ss-icon"
  >
    ðŸ”Ž
  </span>
  zoom
</a>
`;
